### PR TITLE
Add node provider support.

### PR DESCRIPTION
**Pull request by VictorLowther**
This adds the framework to let the Rebar node API drive an external
service.  This is inteded to let Rebar create, detroy, and manage nodes
in external clouds.

### CODE REVIEW

File: rails/app/models/metal_provider.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class MetalProvider < Provider
+
+  audited
```
Comment (galthaus) [inline rails/app/models/metal_provider.rb]: Silly DRY request.  Could inherit all this from (except reboot) from Provider, right?

**Comment by galthaus** [comment]
I'm good with this (even with the DRY issue).

**Comment by galthaus** [comment]
I'll wait for Victor to decide what he wants to do with that.

File: rails/app/models/metal_provider.rb (hunk 1 above)
Comment (VictorLowther) [inline rails/app/models/metal_provider.rb]: Yes, I will fix that.

**Comment by galthaus** [comment]
I'm good.  Are you ready for this to come in?  More coming?

**Comment by VictorLowther** [comment]
This PR is self contained.  There will be others that build on its work.

**Pull request merged**